src/plugin/ComponentManager.h

```diff
@@ -119,24 +119,59 @@ class ComponentManager {
 
     /**
      * @param name The canonical name of a component.
+     * @param consider_remappings If true, component mappings will be honored
+     * if the original component is disabled.
      * @return The component associated with the name or a null pointer if no
      * such component exists.
      */
-    C* Lookup(const std::string& name) const;
+    C* Lookup(const std::string& name, bool consider_remappings = true) const;
 
     /**
      * @param name A component tag.
+     * @param consider_remappings If true, component mappings will be honored
+     * if the original component is disabled.
      * @return The component associated with the tag or a null pointer if no
      * such component exists.
      */
-    C* Lookup(const zeek::Tag& tag) const;
+    C* Lookup(const zeek::Tag& tag, bool consider_remappings = true) const;
 
     /**
      * @param name A component's enum value.
+     * @param consider_remappings If true, component mappings will be honored
+     * if the original component is disabled.
      * @return The component associated with the value or a null pointer if no
      * such component exists.
      */
-    C* Lookup(EnumVal* val) const;
+    C* Lookup(EnumVal* val, bool consider_remappings = true) const;
+
+    /**
+     * Registers a mapping of a component to another one that will be honored
+     * by the `Lookup()` methods if (and only if) the original is currently
+     * disabled.
+     *
+     * @param old The original component tag.
+     * @param new_ The new component tag.
+     */
+    void AddComponentMapping(const zeek::Tag& old, const zeek::Tag& new_) {
+        if ( old != new_ ) {
+            component_mapping_by_src[old] = new_;
+            component_mapping_by_dst[new_] = old;
+        }
+    }
```

Comment:
Does this actually work when stacking multiple EVT `replaces` on top of each other, e.g., in this order have

- `spicy-http` `replaces` `HTTP`, and 
- `my-http` `replace` `HTTP` as well
   - even worse: `my-http` `replaces` `HTTP`

I feel we really need  a stack to model this, indexed by the most basic replacement (in above example: `HTTP`).

Reply:
Sorry, I don't want to go there; at least not for now until we have an actual need/use-case for more complexity here. There aren't that many situations where `replaces` applies in the first place; the standard case is using a Spicy analyzer instead of a built-in, and there are only so many possible instances of that. I don't really see that we need to support chains of analyzers replacing each other; if you have multiple Spicy analyzers doing the same thing, just load the one you need. Otherwise this gets easily really complex and out-of-hand: next thing would be cycle-detection: what if each analyzer disables another one so that eventually have none anymore?

Reply:
That I can do!